expression/constant_propagation.go

```diff
@@ -158,6 +146,51 @@ func (s *propagateConstantSolver) validPropagateCond(cond Expression, funNameMap
 	return nil, nil
 }
 
+// tryToReplaceCond aims to replace all occurances of column 'src' and try to replace it with 'tgt' in 'cond'
+// It returns
+//  bool: if a replacement happened
+//  bool: if 'cond' contains non-deterministic expression
```

Comment:
It seems that  we  can remove this return value?

Reply:
This boolean value is used to avoid some bad case like: `a = b and cast(b, int) < rand()`, for which we cannot propagate a `cast(a, int) < rand()`.

There may be more sophisticated solution, do you have any advise? thanks!

Reply:
ok, never mind